### ISSUE TITLE
Stop profiles when SentrySDK.startWithOptions is not called

### ISSUE THREAD

**Issue by armcknight** [getsentry/sentry-cocoa#3637]
`h`: What if a user never calls `startWithOptions`? Then the tracer and launch profile would run forever if I'm not mistaken. The auto-generated transactions have a timeout of 30 seconds

https://github.com/getsentry/sentry-cocoa/blob/ac8fd5353ed60b08a13a4ad891b7ca374e258f01/Sources/Sentry/SentryTracer.m#L160-L162

but only if they have an idle timeout or `waitForChildren`

https://github.com/getsentry/sentry-cocoa/blob/ac8fd5353ed60b08a13a4ad891b7ca374e258f01/Sources/Sentry/SentryTracer.m#L231-L234

The `SentryTracerConfiguration.defaultConfiguration` doesn't have an idle timeout or `waitForChildren`. One way to fix this would be to enable `waitForChildren`, or to add an extra flag that `isAutoGeneratedTransaction` check.

This is an edge case; I would be OK with fixing it with an upcoming PR.

_Originally posted by @philipphofmann in https://github.com/getsentry/sentry-cocoa/pull/3621#discussion_r1486022257_

**Comment by kahest** [comment]
@armcknight is this still relevant?

**Comment by armcknight** [comment]
@kahest honestly i don't think it's relevant. this is a misconfiguration of the SDK as far as I'm concerned, to call start once and configure launch profiling, and then on the subsequent launch to not call start. Maybe we can message better about this in docs.

**Comment by kahest** [comment]
@armcknight makes sense, feel free to close this or keep it as a reminder to add docs
